xarray/namedarray/_typing.py

```diff
@@ -138,6 +138,14 @@ def __array_function__(
     ) -> Any:
         ...
 
+    @property
+    def imag(self) -> _arrayfunction[_ShapeType_co, Any]:
+        ...
+
+    @property
+    def real(self) -> _arrayfunction[_ShapeType_co, Any]:
+        ...
```

Comment:
Not sure why the numpy strategy of replacing self with a new type doesn't work here.
Oh well, NamedArray has the correct dtype so it's good enough for me.